src/components/pages/City.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from "react-router-dom";
+import { fetchCityWeatherData } from 'actions';
+import PropTypes from 'prop-types';
+import Icon from 'components/reusable/icons/Icon';
+import styles from 'assets/css/styles.scss';
+import temperatureIcon from 'assets/images/icons/thermometer.svg';
+import cloudIcon from 'assets/images/icons/cloudy.svg';
+
+class City extends React.PureComponent {
+  constructor(props){
+    super(props);
+    this.matchProps =
+      this.props.match.params.cityId?
+      this.props.match.params:
+      {cityId:0}; 
+  }
+
+  componentDidMount(){
+    this.props.dispatch(fetchCityWeatherData(this.matchProps.cityId));
```

Comment:
just go with `this.props.dispatch(fetchCityWeatherData(this.props.match.params));`